### ISSUE TITLE
pulumi panics while importing an existing helm release

### ISSUE THREAD

**Issue by jorgeAtNesto** [pulumi/pulumi-kubernetes#1842]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

I'm trying to import an existing helm release (`ingress-nginx`), but pulumi failed to display a preview of the changes to apply and started panicking instead. 

Here's the code snippet that is causing trouble.
```TS
const nginxIngress = new k8s.helm.v3.Release(
  ngReleaseName, 
  {
    chart: 'ingress-nginx'
    namespace: 'default'
    version: '3.20.1'
    values: {
      controller:
      name: controller
      image:
        repository: 'k8s.gcr.io/ingress-nginx/controller'
        tag: 'v0.43.0' 
    ...
  }, 
  {
    import: ngReleaseId,
    protect: true,
    provider: k8sProvider,
  }
);
```
I compared the values I'm using with those utilized to deploy the resource in the first place. There's no mismatch as far as I can see, but even if this was the case, I should get a warning during the preview stage.

Stack trace:

```
pulumi:pulumi:Stack (legacy-test-prod):
    panic: fatal: An assertion has failed
    goroutine 84 [running]:
    github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.failfast(...)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.19.0/go/common/util/contract/failfast.go:23
    github.com/pulumi/pulumi/sdk/v3/go/common/util/contract.Assert(...)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.19.0/go/common/util/contract/assert.go:26
    github.com/pulumi/pulumi/sdk/v3/go/common/util/mapper.(*mapper).EncodeValue(0x14002250fa0, 0x0, 0x0, 0x104948101, 0x0, 0x0, 0x1049e7ca0)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.19.0/go/common/util/mapper/mapper_encode.go:71 +0xd0c
    github.com/pulumi/pulumi/sdk/v3/go/common/util/mapper.(*mapper).EncodeValue(0x14002250fa0, 0x104a259a0, 0x140027fbc80, 0x104948101, 0x104a259a0, 0x140027fbc80, 0x1049e7ca0)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.19.0/go/common/util/mapper/mapper_encode.go:116 +0x1ec
    github.com/pulumi/pulumi/sdk/v3/go/common/util/mapper.(*mapper).EncodeValue(0x14002250fa0, 0x104a259a0, 0x140027fb9e0, 0x104948101, 0x104a259a0, 0x140027fb9e0, 0x1049e7ca0)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.19.0/go/common/util/mapper/mapper_encode.go:116 +0x1ec
    github.com/pulumi/pulumi/sdk/v3/go/common/util/mapper.(*mapper).EncodeValue(0x14002250fa0, 0x104a259a0, 0x140027fbdd0, 0x10485b501, 0x104a259a0, 0x140027fbdd0, 0x1400023bdb0)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.19.0/go/common/util/mapper/mapper_encode.go:116 +0x1ec
    github.com/pulumi/pulumi/sdk/v3/go/common/util/mapper.(*mapper).Encode(0x14002250fa0, 0x1048b4440, 0x1400023bd10, 0x0, 0x14002250fc8, 0x102d74140)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.19.0/go/common/util/mapper/mapper_encode.go:46 +0x3e0
    github.com/pulumi/pulumi/sdk/v3/go/common/util/mapper.Unmap(0x1048b4440, 0x1400023bd10, 0x14000997128, 0x1044666d4, 0x105fb8dc0)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.19.0/go/common/util/mapper/mapper.go:90 +0x94
    github.com/pulumi/pulumi/sdk/v3/go/common/resource.NewPropertyMapRepl(0x1048b4440, 0x1400023bd10, 0x0, 0x0, 0x0)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.19.0/go/common/resource/properties.go:46 +0x30
    github.com/pulumi/pulumi/sdk/v3/go/common/resource.NewPropertyMap(...)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.19.0/go/common/resource/properties.go:39
    github.com/pulumi/pulumi-kubernetes/provider/v3/pkg/provider.(*helmReleaseProvider).serializeImportInputs(...)
        /home/runner/work/pulumi-kubernetes/pulumi-kubernetes/provider/pkg/provider/helm_release.go:912
    github.com/pulumi/pulumi-kubernetes/provider/v3/pkg/provider.(*helmReleaseProvider).Read(0x1400007f500, 0x104ddb3a8, 0x140020282a0, 0x14001fdf900, 0x0, 0x0, 0x6a)
        /home/runner/work/pulumi-kubernetes/pulumi-kubernetes/provider/pkg/provider/helm_release.go:879 +0x884
    github.com/pulumi/pulumi-kubernetes/provider/v3/pkg/provider.(*kubeProvider).Read(0x14000683200, 0x104ddb3a8, 0x140020282a0, 0x14001fdf900, 0x14000683200, 0x10495aea0, 0x14000997898)
        /home/runner/work/pulumi-kubernetes/pulumi-kubernetes/provider/pkg/provider/provider.go:1826 +0x1324
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Read_Handler.func1(0x104ddb3a8, 0x140020282a0, 0x104c2b180, 0x14001fdf900, 0x104c4fb60, 0x105fb6868, 0x104ddb3a8, 0x140020282a0)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.19.0/proto/go/provider.pb.go:2620 +0x7c
    github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1(0x104ddb3a8, 0x14002028030, 0x104c2b180, 0x14001fdf900, 0x140015a0e20, 0x14001ff6f48, 0x0, 0x0, 0x104d90628, 0x140003aef40)
        /home/runner/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x260
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Read_Handler(0x104cf2da0, 0x14000683200, 0x104ddb3a8, 0x14002028030, 0x14001ffb200, 0x1400088f900, 0x104ddb3a8, 0x14002028030, 0x14000fc2bd0, 0x6f)
        /home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.19.0/proto/go/provider.pb.go:2622 +0x12c
    google.golang.org/grpc.(*Server).processUnaryRPC(0x140001e3a40, 0x104df8638, 0x14000622480, 0x14001ff99e0, 0x140008ab470, 0x105f531d8, 0x0, 0x0, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1286 +0x3e8
    google.golang.org/grpc.(*Server).handleStream(0x140001e3a40, 0x104df8638, 0x14000622480, 0x14001ff99e0, 0x0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:1609 +0xa50
    google.golang.org/grpc.(*Server).serveStreams.func1.2(0x1400056e040, 0x140001e3a40, 0x104df8638, 0x14000622480, 0x14001ff99e0)
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:934 +0x94
    created by google.golang.org/grpc.(*Server).serveStreams.func1
        /home/runner/go/pkg/mod/google.golang.org/grpc@v1.38.0/server.go:932 +0x1f8
 
    error: preview failed
```

### Steps to reproduce

1. Given a k8s cluster on GCP, and a nginx-ingress controller deployed using helm and the ingress-nginx chart,
2. Import a new release by creating a new instance of `@pulumi/kubernetes/helm/v3/Release` and adding the import param in the `CustomResourceOptions` (the value must correspond to the id of the Release as defined inside the cluster).
3. run the CLI: `pulumi up`

Expected:  The preview is displayed without errors and allows to identify if there is a mismatch among the actual resource attributes and the args used for importing it.
Actual:  pulumi panics and preview fails.

**Comment by viveklak** [comment]
Looking at the values provided, this seems to be caused by the following options:
```
...
  serviceAccount:
    create: true
    name: null # --> 
  tolerations: []
imagePullSecrets: []
podSecurityPolicy:
  enabled: false
rbac:
  create: true
  scope: false
revisionHistoryLimit: 10
serviceAccount:
  create: true
  name: null # ---> 
...
```

@jorgeAtNesto - it seems these are defaults for the `values.yml` for the chart anyway. Could you remove these entries and try the import? It seems we are a bit more sensitive to these null values and I will look to sanitize the provided values against this.

**Comment by jorgeAtNesto** [comment]
Hi @viveklak, 
Thanks for replying so promptly. 
I was hoping to be able to import the release since we're using version 3.20.1 which needs to be updated. 
That said, I tried a couple of things:
 1. Commenting out problematic fields
 2. Replacing null values with the actual service account name created alongside the deployment. 
 3. Omitting values altogether.
None of which worked 😞 
I provisioned a newer version of the chart (4.0.6) for testing purposes, run the pulumi CLI and managed to import the release. I noted a few differences that might be worth commenting:
```yaml 
  ...
  serviceAccount:
    automountServiceAccountToken: true --> this key is not declared in chart v3.20.1
    create: true
    name: "" --> chart v4.0.6 uses an empty string instead of a null value.
  tolerations: []
imagePullSecrets: []
...
serviceAccount:
  automountServiceAccountToken: true
  create: true
  name: ""
```
So the bug may be somehow related to the chart's version?
